server/src/main/java/org/opensearch/indices/IndicesModule.java

```diff
@@ -281,7 +284,30 @@ protected void configure() {
         bind(RetentionLeaseBackgroundSyncAction.class).asEagerSingleton();
         bind(RetentionLeaseSyncer.class).asEagerSingleton();
         if (FeatureFlags.isEnabled(FeatureFlags.REPLICATION_TYPE)) {
-            bind(SegmentReplicationCheckpointPublisher.class).asEagerSingleton();
+            bind(SegmentReplicationCheckpointPublisher.class).toProvider(CheckpointPublisherProvider.class).asEagerSingleton();
```

Comment:
Since you have an `else` bind statement, you don't need really need a Provider here, and you can retain the `@Inject` constructor